deeppavlov/dataset_iterators/ner_few_shot_iterator.py

```diff
@@ -0,0 +1,149 @@
+
+        self.n_samples = len(self.train)
+
+        if self.remove_not_targets:
+            self._remove_not_target_tags()
+
+        if self.filter_bi:
+            for key in self.data:
+                for n, (x, y) in enumerate(self.data[key]):
+                    self.data[key][n] = [x, [re.sub('(B-|I-)', '', tag) for tag in y]]
+
+        self.tag_map = np.zeros(self.n_samples, dtype=bool)
+        for n, (toks, tags) in enumerate(self.data['train']):
+            if self.filter_bi:
+                self.tag_map[n] = any(self.target_tag == tag for tag in tags if len(tag) > 2)
+            else:
+                self.tag_map[n] = any(self.target_tag == tag[2:] for tag in tags if len(tag) > 2)
+
+        self.marked_nums = None
+        self.unmarked_nums = None
+        self._sample_marked()
+
+    def _sample_marked(self):
+        np.zeros(len(self.data['train']), dtype=bool)
+        n_marked = 0
+        self.marked_mask = np.zeros(self.n_samples, dtype=bool)
+        while n_marked < self.n_train_samples:
+            is_picked = True
+            while is_picked:
+                n = np.random.randint(self.n_samples)
+                if not self.marked_mask[n]:
+                    is_picked = False
+                    self.marked_mask[n] = True
+                    if self.tag_map[n]:
+                        n_marked += 1
+
+        self.marked_nums = np.arange(self.n_samples)[self.marked_mask]
+        self.unmarked_nums = np.arange(self.n_samples)[~self.marked_mask]
+
+    def _remove_not_target_tags(self):
+        if self.remove_not_targets:
+            for key in self.data:
+                for n, (x, y) in enumerate(self.data[key]):
+                    tags = []
+                    for tag in y:
+                        if tag.endswith('-' + self.target_tag):
+                            tags.append(tag)
+                        else:
+                            tags.append('O')
+                    self.data[key][n] = [x, tags]
+
+    def get_instances(self, data_type: str = 'train') -> Tuple[List[List[str]], List[List[str]]]:
+        """Get all data for a selected data type
+
+        Args:
+            data_type (str): can be either ``'train'``, ``'test'``, ``'valid'`` or ``'all'``
+
+        Returns:
+             a tuple of all inputs for a data type and all expected outputs for a data type
+        """
+
+        if data_type == 'train':
+            samples = [self.data[data_type][i] for i in self.marked_nums]
+        else:
+            samples = self.data[data_type][:]
+
+        x, y = list(zip(*samples))
+
+        return x, y
+
+    def gen_batches(self, batch_size: int,
+                    data_type: str = 'train',
+                    shuffle: bool = None) -> Iterator[Tuple[List[List[str]], List[List[str]]]]:
+        if data_type == 'train':
+            x, y = self.get_instances('train')
+        else:
+            x, y = list(zip(*self.data[data_type]))
```

Comment:
can't you just remove this `if` and always `x, y = self.get_instances(data_type)`?